File: 2/2000s/29s/2971/FindPolygonWithTheLargestPerimeter.cs
/*
https://leetcode.com/problems/find-polygon-with-the-largest-perimeter/description/?envType=daily-question&envId=2024-02-15
*/

public class Solution {
    public long LargestPerimeter(int[] nums) {
        Array.Sort(nums);
        int sz=nums.Length;
        for(int i=sz-1; i>=2; --i){
            long curval=nums[i], cursm=0;
            for(int j=0; j<i; ++j){
                cursm+=nums[j];
            }
            if(cursm>curval) return cursm+curval;
        }
        return -1;
    }
}

File: 2/2000s/29s/2971/FindPolygonWithTheLargestPerimeter.py
# https://leetcode.com/problems/find-polygon-with-the-largest-perimeter/description/?envType=daily-question&envId=2024-02-15

class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        sz=len(nums)
        for i in range(sz-1, -1, -1):
            curval=nums[i]
            cursm=0 
            for j in range(0, i, 1):
                cursm+=nums[j]
            if cursm>curval and (i-1)+1>=2:
                return cursm+curval
        return -1